### PR TITLE
wolfictl vex sbom subcommand

### PR DESCRIPTION
**Pull request by puerco**
This PR introduces the `wolfictl vex sbom` subcommand. This enables wolfictl to generate vex documents from SBOMs containing wolfi packages. 

Follow-up to #11 , ~needing a rebase after merging that one~ **Update:** rebased :white_check_mark: .

/cc @luhring 

```
wolfi ctrl vex sbom --help
wolfictl vex sbom: Generate a VEX document from wolfi packages listed in an SBOM
		
The vex sbom subcommand generates VEX documents describing how vulnerabilies
impact Wolfi packages listed in an SBOM. This subcommand reads SPDX SBOMs and
will recognize and capture all packages identified as Wolfi OS components 
by its purl. For example, if an SBOM contains a package with the following
purl:

	pkg:apk/wolfi/curl@7.87.0-r0
	
wolfictl will read the melange configuration file that created the package and
create a VEX document containing impact assessments in its advisories and
secfixes.

wolfictl will read the melange config files from an existing wolfi-dev/os clone
or, if not specified, it will clone the repo for you.

Usage:
  wolfictl vex sbom [flags] sbom.spdx.json

Examples:
wolfictl vex sbom --author=joe@doe.com sbom.spdx.json

Flags:
      --author string   author of the VEX document
  -h, --help            help for sbom
      --repo string     path to a local clone of the wolfi-dev/os repo
      --role string     role of the author of the VEX document

```

### CODE REVIEW

**Comment by puerco** [comment]
OK, the initial implementation of this is done!

Here is an example of a run using the SBOM attached to this PR. I'm attaching the produced VEX document too. @luhring as you'll see, the installed packages (according to the SBOM) are included in the subcomponents list as we discussed. We can eliminate them but I think it makes sense to keep them there. 

### Invocation
```
wolfi vex sbom  pkg/vex/testdata/git.spdx.json 
```

### Output:
```json
{
  "id": "merged-vex-e45086b18fbc879c482f6c2a304f3909a6e3a827fcebaefafc9a937df7600b9a",
  "format": "text/vex",
  "author": "",
  "role": "",
  "timestamp": "2022-12-28T22:07:05.495089156-06:00",
  "statements": [
    {
      "vulnerability": "CVE-2018-25032",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/zlib@1.2.13-r1?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2020-8927",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/libbrotlicommon1@1.0.9-r0?distro=wolfi",
        "pkg:apk/wolfi/libbrotlidec1@1.0.9-r0?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2020-8927",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/libbrotlicommon1@1.0.9-r0?distro=wolfi",
        "pkg:apk/wolfi/libbrotlidec1@1.0.9-r0?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-29458",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/ncurses@6.3-r2?distro=wolfi",
        "pkg:apk/wolfi/ncurses-terminfo-base@6.3-r2?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-29458",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/ncurses@6.3-r2?distro=wolfi",
        "pkg:apk/wolfi/ncurses-terminfo-base@6.3-r2?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-37434",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/zlib@1.2.13-r1?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-39046",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/glibc@2.36-r3?distro=wolfi",
        "pkg:apk/wolfi/glibc-locale-posix@2.36-r3?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-39046",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/glibc@2.36-r3?distro=wolfi",
        "pkg:apk/wolfi/glibc-locale-posix@2.36-r3?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-40674",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/expat@2.5.0-r0?distro=wolfi"
      ],
      "status": "fixed"
    },
    {
      "vulnerability": "CVE-2022-43680",
      "products": [
        "pkg:oci/git@sha256:54a88f29b889d82e57712206973db99089caf4074232bb16df8c72605aaaa410?arch=amd64&mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson&os=linux"
      ],
      "subcomponents": [
        "pkg:apk/wolfi/expat@2.5.0-r0?distro=wolfi"
      ],
      "status": "fixed"
    }
  ]
}

```

**Comment by puerco** [comment]
Example generated VEX document attached here: [image.vex.json.gz](https://github.com/wolfi-dev/wolfictl/files/10317565/image.vex.json.gz)

File: pkg/vex/vex.go [hunk 1]
```diff
@@ -48,6 +182,121 @@ func FromPackageConfiguration(vexCfg Config, buildCfg ...*build.Configuration) (
...
+		descPurl := ""
+		for i := range sbom.Packages {
+			if sbom.Packages[i].ID != elementID {
+				continue
+			}
+			for _, ref := range sbom.Packages[i].ExternalRefs {
+				if ref.Type == "purl" {
+					descPurl = ref.Locator
+				}
+			}
```
Comment (luhring) [inline pkg/vex/vex.go:275]: Just so I understand, does this mean "last purl wins"? I'm guessing that's fine

File: pkg/cli/vex.go [hunk 2]
```diff
@@ -38,12 +38,18 @@ more about the VEX tooling powering wolfictk see: https://github.com/chainguard-
 	}
 
 	addPackages(cmd)
-
+	addSBOM(cmd)
 	return cmd
 }
 
+var vexCfg = vex.Config{
+	Distro:     "wolfi",
```
Comment (luhring) [inline pkg/cli/vex.go:46]: (non-blocking for this PR) Maybe we should get rid of the `distro` field here. Now that we're working in `wolfictl`, there shouldn't be functionality for non-Wolfi distros

**Review by luhring — APPROVED** [review]
Very nice!

File: pkg/vex/vex.go (hunk 1 above)
Comment (puerco) [reply pkg/vex/vex.go:275]: Yes, although we filter purls to only those namespaced to Wolfi (the distro param) so it should always be just one. Perhaps we can spare the cycle there.

File: pkg/cli/vex.go (hunk 2 above)
Comment (puerco) [reply pkg/cli/vex.go:46]: Ok, sounds good. And going a little bit beyond, (thinking about the ?distro=qualifier) maybe we should prepare and design these tools to work in a scenario where we have more than one flavor of Wolfi.

**Pull request merged**